### PR TITLE
Add objc clang format

### PR DESCRIPTION
**Pull request by qqaatw**
The main PR is #7643. Submitting this PR first to get the CI workflows in the main PR run normally without further approval.

cc @NicolasHug @pmeier.

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -60,9 +60,6 @@ MacroBlockBegin: ''
 MacroBlockEnd:   ''
 MaxEmptyLinesToKeep: 1
 NamespaceIndentation: None
-ObjCBlockIndentWidth: 2
-ObjCSpaceAfterProperty: false
-ObjCSpaceBeforeProtocolList: false
```
Comment (NicolasHug) [inline .clang-format:65]: Do we actually want to remove those from here @qqaatw ? Isn't the newly added section below exclusively related to objective-C?

File: .clang-format [hunk 2]
```diff
@@ -85,4 +82,11 @@ SpacesInSquareBrackets: false
...
 UseTab:          Never
-...
+---
+Language: ObjC
+ColumnLimit: 120
+AlignAfterOpenBracket: Align
+ObjCBlockIndentWidth: 2
+ObjCSpaceAfterProperty: false
+ObjCSpaceBeforeProtocolList: false
+...
```
Comment (NicolasHug) [inline .clang-format:92]: I think / hope this should resolve both CI failures from https://github.com/pytorch/vision/actions/runs/5312942594/jobs/9618529905?pr=7677

```suggestion
...

```

File: .clang-format (hunk 1 above)
Comment (NicolasHug) [reply .clang-format:65]: Oh my bad, I completely missed that these are all `ObjC...` keys already anyway

**Review by NicolasHug — APPROVED** [review]
LGTM but the yaml checker is tripping. @pmeier any insight? Maybe we should get rid of it if it's becoming unreliable

**Comment by pmeier** [comment]
Kinda surprised that it is picking `.clang-format` as viable target in the first place. I'll have a look.

**Comment by pmeier** [comment]
According to the [`clang-format` docs](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#configuring-style-with-clang-format):

> The `.clang-format` file uses YAML format:

The hook is configured to [trigger on `yaml` files](https://github.com/pre-commit/pre-commit-hooks/blob/bb4e628bb475392bed0736990db6f33009bb8169/.pre-commit-hooks.yaml#L92-L97). The [`pre-commit` docs](
https://pre-commit.com/#filtering-files-with-types) state

> The tags are discovered through a set of heuristics by the [identify](https://github.com/pre-commit/identify) library. identify was chosen as it is a small portable pure python library.

And indeed

```bash
$ identify-cli .clang-format
["file", "non-executable", "text", "yaml"]
```

In the YAML specification, [each `---` indicates a separate document](https://yaml.org/spec/1.2.2/#structures). Meaning, here we have two documents in a single file. This is what the hook is complaining about. However, this is not only correct, but also the way to go according to the `clang-format` documentation.

Thus, we have three options:

1. Ignore `.clang-format` in the `check-yaml` hook
2. Add the [`--allow-multiple-documents`](https://github.com/pre-commit/pre-commit-hooks/blob/bb4e628bb475392bed0736990db6f33009bb8169/pre_commit_hooks/check_yaml.py#L42-L44) option to the hook
3. Disable the hook completely

My preference is 2 > 1 > 3.

**Comment by qqaatw** [comment]
@NicolasHug @pmeier Thanks. Added the option.

**Pull request merged**